crates/ironrdp-web/src/session.rs

```diff
@@ -492,18 +477,120 @@ impl Session {
                     }
                     ActiveStageOutput::PointerDefault => {
                         let _ret = self
-                            .show_pointer_callback
-                            .call0(&self.show_pointer_callback_context)
-                            .map_err(|e| anyhow::Error::msg(format!("show pointer callback failed: {e:?}")))?;
+                            .set_cursor_style_callback
+                            .call3(
+                                &self.set_cursor_style_callback_context,
+                                &JsValue::from_str("default"),
+                                &JsValue::UNDEFINED,
+                                &JsValue::UNDEFINED,
+                            )
+                            .map_err(|e| anyhow::Error::msg(format!("set cursor style callback failed: {e:?}")))?;
                     }
                     ActiveStageOutput::PointerHidden => {
                         let _ret = self
-                            .hide_pointer_callback
-                            .call0(&self.hide_pointer_callback_context)
-                            .map_err(|e| anyhow::Error::msg(format!("hide pointer callback failed: {e:?}")))?;
+                            .set_cursor_style_callback
+                            .call3(
+                                &self.set_cursor_style_callback_context,
+                                &JsValue::from_str("none"),
+                                &JsValue::UNDEFINED,
+                                &JsValue::UNDEFINED,
+                            )
+                            .map_err(|e| anyhow::Error::msg(format!("set cursor style callback failed: {e:?}")))?;
                     }
                     ActiveStageOutput::PointerPosition { .. } => {
-                        // Not applicable for web
+                        // Not applicable for web.
+                    }
+                    ActiveStageOutput::NativePointerUpdate(pointer) => {
+                        // Maximum allowed cursor size for browsers is 32x32, because bigger sizes
+                        // will cause the following issues:
+                        // - 128x128 and bigger cursors are not supported in browsers.
+                        // - 32x32 and bigger cursors will default to the system cursor if their
+                        //   sprite does not fit in the browser's viewport, introducing anabrupt
+                        //   cursor style changes when the cursor is moved to the edge of the
+                        //   browser window.
+                        //
+                        // Therefore, we need to scale the cursor sprite down to 32x32 if it is
+                        // bigger than that.
+                        const MAX_CURSOR_SIZE: u16 = 32;
+                        // INVARIANT: 0 < scale <= 1.0
```

Comment:
Note: those are additional invariants that are useful when reasoning about the following code:
```rust
// INVARIANT: pointer.width * scale = MAX_CURSOR_SIZE
// INVARIANT: pointer.height * scale = MAX_CURSOR_SIZE
```